tools/integration-test/src/tests/client_settings.rs

```diff
@@ -0,0 +1,125 @@
+use std::time::Duration;
+
+use ibc::core::ics02_client::trust_threshold::TrustThreshold;
+
+use ibc::clients::ics07_tendermint::client_state::ClientState as TendermintClientState;
+use ibc::core::ics02_client::client_state::AnyClientState;
+use ibc::Height;
+use ibc_relayer::chain::client::ClientSettings;
+use ibc_relayer::chain::cosmos;
+
+use ibc_test_framework::prelude::*;
+
+/// A test to exercise default foreign client settings.
+#[test]
+fn test_client_defaults() -> Result<(), Error> {
+    run_binary_chain_test(&ClientDefaultsTest)
+}
+
+/// A test to exercise customization of foreign client settings.
+#[test]
+fn test_client_settings() -> Result<(), Error> {
+    run_binary_chain_test(&ClientSettingsTest)
+}
+
+struct ClientDefaultsTest;
+
+struct ClientSettingsTest;
+
+struct SettingsTestOverrides;
+
+impl TestOverrides for ClientDefaultsTest {
+    fn modify_relayer_config(&self, config: &mut Config) {
+        config.chains[0].clock_drift = Duration::from_secs(3);
+        config.chains[0].max_block_time = Duration::from_secs(5);
+        config.chains[0].trusting_period = Some(Duration::from_secs(120_000));
+        config.chains[0].trust_threshold = TrustThreshold::new(13, 23).unwrap().try_into().unwrap();
+
+        config.chains[1].clock_drift = Duration::from_secs(6);
+        config.chains[1].max_block_time = Duration::from_secs(15);
+        config.chains[1].trusting_period = Some(Duration::from_secs(340_000));
+        config.chains[1].trust_threshold = TrustThreshold::TWO_THIRDS.try_into().unwrap();
+    }
+}
+
+impl BinaryChainTest for ClientDefaultsTest {
```

Comment:
I think it would be clearer to have a `BinaryNodeTest` for this case, with the foreign clients being created manually with the custom client settings you want. Basically the test should call `ForeignClient::new()` or `ForeignClient::restore()` directly, and then call `ForeignClient::build_create_client_and_send()` with the custom client settings, because this is the actual method we want to test here.

For this particular test, it is more of testing that the test framework is correctly initializing the foreign clients with the wanted settings. We could still leave the test here for testing the correctness of the test framework itself.

I think it is also worth writing some simple tests to demonstrate why we need a `SettingsTestOverrides`. For example, is having the ability to override the `max_clock_drift` setting helps us write new integration tests that we couldn't before? If the tests are too complicated to write in this PR, then we should file a new issues to follow up on writing these tests.

Reply:
Good points, the tests could exercise the API directly indeed rather than deferring it to the framework.

As for `ClientSettingsOverride`, I think the ability to tweak the pre-created clients could be useful, and it does not get in the way of writing any tests that don't need this feature. The long trait bound lists in the framework code itself could become unwieldy, but they could be made more manageable with helper traits if needed. One thing this override trait cannot address is n-ary tests, but the same problem exists for the other override traits.

Reply:
Ok, so do you think it is unnecessary to write another test using `BinaryNodeTest` to test `ForeignClient::build_create_client_and_send()` directly?

> One thing this override trait cannot address is n-ary tests, but the same problem exists for the other override traits.

For the case of ports override, we use nested arrays to provide a matrix of overrides. But I think we can just omit that for now and only add them when it is needed.

Reply:
@romac says custom client settings could be useful to test misbehavior detection.

Reply:
Actually I meant writing an additional test to test the function directly. But anyway it's a nit and doesn't matter too much.